src/v1.js

```diff
@@ -586,6 +587,59 @@ builder.declare({
   res.reply(roles.map(r => roleToJson(r, this)));
 });
 
+/** List role Ids **/
+builder.declare({
+  method:     'get',
+  route:      '/roleids/',
+  query: {
+    continuationToken: /./,
+    limit: /^[0-9]+$/,
+  },
+  name:       'listRoleIds',
+  input:      undefined,
+  output:     'list-role-ids-response.yml',
+  stability:  'stable',
+  title:      'List Role IDs',
+  description: [
+    'If no limit is given, the roleIds of all roles are returned. Since this',
+    'list may become long, callers can use the `limit` and `continuationToken`',
+    'query arguments to page through the responses.',
+  ].join('\n'),
+}, async function(req, res) {
+  let hashids = new Hashids();
+  let continuationToken = parseInt(req.query.continuationToken, 10);
+  let limit = parseInt(req.query.limit, 10);
+  let response = {};
+
+  // Load all roles
+  let roles = await this.Roles.get();
```

Comment:
Silly question but why are not doing `Roles.scan({}, opts);`?

Reply:
Roles is a single blob in Azure blob storage, rather than an Azure table with multiple rows.

With parameterized roles, there are combinations of roles that are invalid.  We need to prevent such combinations from being written to the role storage.  It's not possible to make atomic updates to a entire Azure table, so if we were using table storage, race conditions would permit writing invalid combinations to the table.  So, we switched to a single blob that we can read and write atomically.